File: meta-anime-frontend/src/types/anime.ts
export interface Anime {
    animeId: number,
    title: AnimeTitles,
    coverImage: string,
    averageScore: number,
    mappings: [AnimeMapping]
}

export interface AnimeMapping {
    mappingId: number,
    sourcePlatform: string,
    platformId: string,
    rawScore: number,
}

export interface AnimeTitles {
    titleNative: string,
    titleRomaji: string,
    titleEn: string,
    titleCn: string,
}

File: meta-anime-backend/src/main/java/com/iccues/metaanimebackend/entity/Anime.java
package com.iccues.metaanimebackend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Anime {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long animeId;

    @Embedded
    AnimeTitles title;

    String coverImage;

    LocalDate startDate;

    Double averageScore;

    @OneToMany(mappedBy = "anime",
            cascade = {CascadeType.PERSIST, CascadeType.MERGE},
            fetch = FetchType.LAZY)
    List<Mapping> mappings = new ArrayList<>();

    public void addMapping(Mapping mapping) {
        this.mappings.add(mapping);
        mapping.setAnime(this);
    }

    public void removeMapping(Mapping mapping) {
        this.mappings.remove(mapping);
        mapping.setAnime(null);
    }
}
